great_expectations/metrics/metric_results.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, Literal, NamedTuple, Union
+
+from great_expectations.compatibility.pydantic import BaseModel, validator
+
+IDDictID = Union[str, tuple[()]]
+
+
+class MetricConfigurationID(NamedTuple):
+    metric_name: str
+    metric_domain_kwargs_id: IDDictID
+    metric_value_kwargs_id: IDDictID
+
+
+class _MetricResult(BaseModel):
+    id: MetricConfigurationID
+    error: bool
+
+
+class _SuccessfulMetricResult(_MetricResult):
+    error: Literal[False] = False
+
+
+class InvalidMetricError(TypeError):
+    def __init__(self, expected_metric: str, actual_metric: str):
+        super().__init__(
+            f"Invalid metric: expected {expected_metric} but received {actual_metric}."
+        )
+
+
+class TableColumns(_SuccessfulMetricResult):
```

Comment:
The names of these subclasses mirror the names of the metrics themselves. I think we need a different convention for the result classes if we even need the subclasses at all.